webrender/src/frame.rs

```diff
@@ -67,15 +67,23 @@ impl NestedDisplayListInfo {
 
     fn convert_scroll_id_to_nested(&self, id: &ClipId) -> ClipId {
         if id.is_root_scroll_node() {
-            self.scroll_node_id
+            if id.pipeline_id() == self.scroll_node_id.pipeline_id() {
+                self.scroll_node_id
+            } else {
+                *id
+            }
         } else {
             self.convert_id_to_nested(id)
```

Comment:
Why don't we skip the conversion for non-root scroll nodes as well? If that's the right thing to do, we can just do an early return from this method.